server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -236,6 +238,13 @@ private static void doValidateCodecSettings(final String codec) {
         Property.Dynamic
     );
 
+    public static final Setting<Boolean> INDEX_USE_COMPOUND_FILE = Setting.boolSetting(
+        "index.use_compound_file",
+        true,
+        Property.IndexScope,
+        Property.Dynamic
```

Comment:
How exactly does this behave with a dynamic setting? The setting will only take effect when an IndexWriter is created, which means that existing index writers will continue operating with the old setting until something forces a new index writer to be created. You could avoid this potentially confusing scenario by not making this dynamic unless it is truly required to be dynamic.

Reply:
My thinking was that at any point we should be able to set the newly written segments and merges to start behaving differently (with/without compound file).
However, I agree that right now the behavior is not entirely clear because there is nothing forcing a recreation of a new index writer when this config changes. Therefore it might happen in a weird and unpredictable way. Will move this to static for now and can change when explicitly forcing a new index writer creation.

Reply:
Sounds good. I did find a similar setting that is dynamic and it [documents the complexity](https://github.com/opensearch-project/OpenSearch/blob/f3d2beee637f63e38c8f26dbcee9f2a82f9c87b6/server/src/main/java/org/opensearch/index/engine/EngineConfig.java#L226-L230). However, I agree making it static is the simplest approach and we can revisit in the future if necessary.